dex/src/matching.rs

```diff
@@ -573,20 +588,22 @@ impl<'ob> OrderBookState<'ob> {
                 insert_result.unwrap();
             }
         } else {
-            to_release.unlock_coin(unfilled_qty);
-            let out = Event::new(EventView::Out {
-                side: Side::Ask,
-                release_funds: false,
-                native_qty_unlocked: unfilled_qty * coin_lot_size,
-                native_qty_still_locked: 0,
-                order_id,
-                owner,
-                owner_slot,
-                client_order_id: NonZeroU64::new(client_order_id),
-            });
-            event_q
-                .push_back(out)
-                .map_err(|_| DexErrorCode::EventQueueFull)?;
+            if owner != solana_program::system_program::ID.to_aligned_bytes() {
```

Comment:
Why is this code asymmetric from the `Bid` side? (There's no `else` clause for Bid)

Reply:
Yeah - Bid and Ask have slightly different code structures. From what I can tell, they're basically doing the same thing.